node_modules/oae-authentication/tests/test-cookies.js

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright 2015 Apereo Foundation (AF) Licensed under the
+ * Educational Community License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may
+ * obtain a copy of the License at
+ *
+ *     http://opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+var _ = require('underscore');
+var assert = require('assert');
+
+var RestAPI = require('oae-rest');
+var TestsUtil = require('oae-tests');
+
+var AuthenticationAPI = require('oae-authentication');
+
+describe('Authentication', function() {
```

Comment:
I don't think this is necessary. Express will always set some cookie on a cookieless request, even if the user isn't authenticated. We have tests that ensure that failed authentications don't result in an authenticated session, so that should suffice.